scripts/jenkins_bootstrap.py

```diff
@@ -0,0 +1,333 @@
+  --jenkins-user <username>         Jenkins username.
+  --jenkins-password <password>     Jenkins Password.
+  -u, --url <url>                   Jenkins URL [default: http://localhost:8080]
+  -r, --retries <int>               Retries for requests [default: 12]
+  -e, --encoding <type>             Encoding type supports text or html [default: text]
+  -p, --parameters <data>           Comma separated job parameters i.e. a=1,b=2
+  -d, --debug                       Print debug info
+  -h, --help                        Show this screen and exit.
+"""
+
+from docopt import docopt
+from time import sleep
+from urllib3.util.retry import Retry
+from re import search
+
+import requests
+import logging
+import atexit
+
+
+def binary_backoff(retries):
+    """
+    Increment by a multiplier based off of number of retries made
+
+    Returns: <int> which resembles number of seconds used to sleep
+    """
+    # Try (#)              | 1 | 2 | 3 | 4 | 5  | 6  | 7  | 8   | 9   | 10  | 11   | 12   | 13   ...
+    # Backoff factor 2 (s) | 1 | 2 | 4 | 8 | 16 | 32 | 64 | 128 | 256 | 512 | 1024 | 2056 | 4096 ...
+    backoff_factor = 2
+    return backoff_factor * ( 2 ** (retries - 1))
+
+
+class Trigger:
+
+    def __init__(self, arguments):
+
+        # Configure retries for http(s) requests
+        adapter = requests.adapters.HTTPAdapter(
+            max_retries=Retry(
+                # Retry total defaults to 12
+                total=ARGUMENTS['--retries'][0],
+                status_forcelist=[429, 500, 502, 503, 504],
+                method_whitelist=["HEAD", "GET", "OPTIONS"],
+                # Binary backoff where timeout = {backoff factor} * (2 ** ({current retries} - 1))
+                backoff_factor=2
+            )
+        )
+
+        # Create request session and mount adapters
+        self.http = requests.Session()
+        self.http.mount("https://", adapter)
+        self.http.mount("http://", adapter)
+
+        # Handle Parameters with docopt
+        self.arguments = arguments
+        self.debug = arguments['--debug']
+        self.job = arguments['--job'][0]
+        self.user = arguments['--jenkins-user']
+        self.password = arguments['--jenkins-password']
+        self.url = arguments['--url'][0]
+        self.encoding = arguments['--encoding'][0]
+        self.retries = int(arguments['--retries'][0])
+        self.crumb = self.fetch_crumb()
+
+        if self.encoding.lower() in ["html", "text"]:
+            self.encoding = self.encoding.title()
+        else:
+            raise ValueError(f"{self.encoding} is not a valid encoding only support 'text', 'html'")
+
+        if arguments['--parameters']:
+            try:
+                self.parameters = dict(u.split("=") for u in arguments['--parameters'][0].split(","))
+            except ValueError as err:
+                raise Exception("Your parameters should be in key=value format separated by ; for multi value i.e. x=1,b=2") from err
```

Comment:
"separated by ; for multi value i.e. x=1,b=2"

Wouldn't this imply the come should be a semi-colon, or the semi-colon, a comma?

Reply:
Good catch. What did you intend to have when you wrote the original script? `;` or `,` ?

Reply:
I believe commas.

Reply:
Thanks, I'll change it momentarily.